### PR TITLE
Add ability to view specific challenge

### LINKED ISSUE #52: As a user of GeoHunt, I want to be able to see a detail view of challenges so that I have more details

### LINKED ISSUE #57: Create challenge page with details and photos

### LINKED ISSUE #58: Create a zoomable image view when the user click on the image

### LINKED ISSUE #59: Link the challenge view with the database

### LINKED ISSUE #61: Add test for the challenge view an the zoomable image view

### PR DESCRIPTION
**Pull request by BoyeGuillaume**
## Changes:
* Improve rememberLazyRef API by adding the new `FetchComponent` and `AsyncImage`
* Add the ability to view specific challenges (given the id) and link this mechanism to the navigation
* Enable to view certain image full screen with a zoomable view and link this mechanism to the navigation
* Implement basic skeleton of `getNearbyChallenge`
* Add format utilities for pattern `%prefix% %duration% ago` (eg. `"published 5 days ago"`)
* Add format utilities for int suffix (eg. `"10k"` instead of `"1486"`)

## TODO:
* Link claims to the database
* Link profile picture to the database

## Issues:

Closes #61, closes #59, closes #58, closes #57 and closes #52.

### CODE REVIEW

**Comment by Maeeen** [comment]
> Add format utilities for pattern %prefix% %duration% ago (eg. "published 5 days ago")

Why not using [RelativeDateTimeFormatter](https://developer.android.com/reference/android/icu/text/RelativeDateTimeFormatter)? I know that I will maybe open up a long debate but instead of re-inventing the wheel, try to use already tested and available API. Please take into account that i18n is a thing.

> Add format utilities for int suffix (eg. "10k" instead of "1486")

I think this should be more properly handled in a i18n package. Not every language is such prefixes (I think of japanese and chinese where especially for japanese, numbers are handled by 4 digits, not 3 digits and special characters exists to denote such numbers)

Overall, good job 👍

**Comment by BoyeGuillaume** [comment]
> > Add format utilities for pattern %prefix% %duration% ago (eg. "published 5 days ago")
> 
> Why not using [RelativeDateTimeFormatter](https://developer.android.com/reference/android/icu/text/RelativeDateTimeFormatter)? I know that I will maybe open up a long debate but instead of re-inventing the wheel, try to use already tested and available API. Please take into account that i18n is a thing.
> 
> > Add format utilities for int suffix (eg. "10k" instead of "1486")
> 
> I think this should be more properly handled in a i18n package. Not every language is such prefixes (I think of japanese and chinese where especially for japanese, numbers are handled by 4 digits, not 3 digits and special characters exists to denote such numbers)
> 
> Overall, good job 👍

For your first comment, I have checked the API and it seems the api is very crude. You basically need to implement everything yourself. Check [this link](https://developer.android.com/reference/android/icu/text/RelativeDateTimeFormatter) for the API. I think implementing this feature using the RelativeDateTime formatter would be harder than the current implementation (Even if reinventing the wheel).

For your second comment, you are totally right. However I think that if we really want to speak local, it would require to rewrite most of our application to support language such as arab where text should be aligned right. As such I though we could let this as a future user story. Right now I believe we should focus our effort to have a working prototype, even though only supporting language structurally close to english.

File: app/src/main/java/com/github/geohunt/app/ui/components/challenge/ClaimCard.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            .fillMaxWidth()
+            .aspectRatio(1.8f, false)
+            .padding(5.dp)
+    ) {
+        FetchComponent(
+            lazyRef = { claimRef },
+            modifier = Modifier.fillMaxSize()
+        ) { claim ->
+            Column {
+                Box(
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/challenge/ClaimCard.kt:41]: Maybe some refactoring, that composable is really huge.

File: app/src/main/java/com/github/geohunt/app/ui/components/challenge/MainUserView.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                            textAlign = TextAlign.Left,
+                            modifier = Modifier
+                                .align(Alignment.Bottom)
+                                .padding(2.dp)
+                                .wrapContentSize(unbounded = true)
+                        )
+
+                        Spacer(modifier = Modifier.weight(1f))
+
+                        LabelledIcon(
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/challenge/MainUserView.kt:65]: Really huge again, please refactor.

File: firebase-debug.log [hunk 3]
```diff
@@ -0,0 +1,72 @@
+[debug] [2023-03-31T08:11:41.180Z] ----------------------------------------------------------------------
```
Comment (Maeeen) [inline firebase-debug.log:1]: This file be removed

**Review by Maeeen — DISMISSED** [review]

**Review by Maeeen — DISMISSED** [review]

**Review by Maeeen — APPROVED** [review]

**Pull request merged**